miner/worker.go

```diff
@@ -345,6 +346,7 @@ func (miner *Miner) prepareWork(genParams *generateParams, witness bool) (*envir
 		common.CurrencyAllowlist(env.feeCurrencyContext.ExchangeRates),
 		header,
 	)
+	env.blockingAllowed = miner.feeCurrencyBlocklist.GetEnableFilterStatus()
```

Comment:
is this done for every block?

Reply:
Yep, env is constructed per block at line 331

Reply:
Yes, every time a new payload is requested to be built by the engine-api a new `environment` is created